app/(api)/_datalib/auth/verifyAuthToken.ts

```diff
@@ -0,0 +1,16 @@
+import jwt, { Secret } from 'jsonwebtoken';
+import HttpError from '@utils/response/HttpError';
+
+export async function verifyAuthToken(token: string) {
+  try {
+    const verifiedPayload = jwt.verify(token, process.env.JWT_SECRET as Secret);
+
+    if (!verifiedPayload) {
+      throw new HttpError('Unauthorized. Invalid token.');
+    }
+
+    return { ok: true, body: verifiedPayload };
```

Comment:
return `{ ok: true, body: verifiedPayload, error: null }`
It's nice to keep a similar structure for both error and success.

for error: return `{ ok: true, body: null, error: error_stuff }`

Reply:
Updated changes!